get.go

```diff
@@ -31,16 +32,16 @@ type Getter interface {
 	// The directory may already exist (if we're updating). If it is in a
 	// format that isn't understood, an error should be returned. Get shouldn't
 	// simply nuke the directory.
-	Get(string, *url.URL) error
+	Get(context.Context, *Request) error
 
 	// GetFile downloads the give URL into the given path. The URL must
 	// reference a single file. If possible, the Getter should check if
 	// the remote end contains the same file and no-op this operation.
-	GetFile(string, *url.URL) error
+	GetFile(context.Context, *Request) error
 
 	// ClientMode returns the mode based on the given URL. This is used to
 	// allow clients to let the getters decide which mode to use.
-	ClientMode(*url.URL) (ClientMode, error)
+	ClientMode(context.Context, *url.URL) (ClientMode, error)
```

Comment:
This function seems more `InferURLMode` or `ModeForURL` ?  Also, consider renaming `ClientMode` to `RequestMode`?

Reply:
I see this is addressed in the follow up PRs: https://github.com/hashicorp/go-getter/pull/232/commits/3d28ee4ed4f05246a41a97cc96b3a351e6125aa3 .  Ignore this - thanks!